File: src/Java/src/ListItemCounter.java
import java.util.Arrays;

public class ListItemCounter {
  public static int listItemCounter(int[] list) {
    final int MID = list.length / 2;
    if (list.length == 1) {
      return 1;
    }
    return listItemCounter(Arrays.copyOfRange(list, 0, MID))
        + listItemCounter(Arrays.copyOfRange(list, MID, list.length));
  }
}

File: src/Python/src/list_item_counter.py
import math
def list_item_counter(list: list[int]) -> int:
  if (len(list) != 1):
    MID = math.floor(len(list) / 2);
    LEFT = list[0: MID]
    RIGHT = list[MID:]
    
    return list_item_counter(LEFT) + list_item_counter(RIGHT)
  return 1